src/Model/Serializer/ValueObject/MediaObject/VideoNormalizer.php

```diff
@@ -34,6 +34,10 @@ final class VideoNormalizer implements NormalizerInterface
             'embed' => self::YOUTUBE_EMBED,
             'name' => self::YOUTUBE_NAME,
         ],
+        11 => [
+            'embed' => self::YOUTUBE_EMBED,
+            'name' => self::YOUTUBE_NAME,
+        ],
```

Comment:
They are the "capture groups" from the regex. Every `()` in the regex is a capture group, which you can get back in the results of the regex. This particular regex has a lot of them because the URL patterns are different depending on the service (YouTube <> Vimeo) and the exact URL that is given. We need to combine it in one regex though so that they fit in JSON schema where we can only define one regex pattern on a particular property.

Reply:
It's maybe more clear in a tool like https://www.phpliveregex.com/#tab-preg-match-all

If you enter as regex: `^http(s?):\/\/(www\.)?((youtube\.com\/watch\?v=([^\/#&?]*))|(vimeo\.com\/([^\/#&?]*))|(youtu\.be\/([^\/#&?]*))|(youtube.com\/embed\/([^\/#&?]*)))`

And as search strings:
```
https://www.youtube.com/watch?v=hHaBhlGQA1w
https://www.youtube.com/embed/hHaBhlGQA1w
```

You will see that the "video id" for example will be found in different indexes in the array of `matches` depending on the exact URL structure. This is because the video id is located in multiple capture groups of the regex, one for each URL pattern.

![Screen Shot 2023-03-31 at 15 34 29](https://user-images.githubusercontent.com/959026/229134401-2dffd342-570d-49a5-95e8-27e8590798ea.png)

Reply:
(Note that this is not unique to PHP but applies to regexes in most programming languages)

Reply:
And to clarify this particular code, we can detect what kind of URL is being used based on which capture group is present or not. For example if capture group 11 is found, it's an embed URL because that capture group can only be present if the URL matches the pattern of an embed URL.